include/seqan3/core/type_traits/iterator.hpp

```diff
@@ -92,7 +92,7 @@ struct rvalue_reference<it_t>
  * \tparam it_t The type you wish to query; must model std::weakly_incrementable.
  */
 template <std::weakly_incrementable it_t>
-struct difference_type<it_t>
+struct SEQAN3_DEPRECATED_310 difference_type<it_t>
```

Comment:
@smehringer  This has absolutely no effect. Good catch!

@wvandertoorn  can you remove this? Thank you!

Reply:
But we want to deprecate this? Why remove it?

Reply:
because
* this PR makes only the changes and the deprecation comes in a later PR (we discussed that in the strategy meeting)
* this deprecation is only on the partial specialisation and it has no effect what-so-ever (attributes are ignored on partial specialisations). The deprecation must happen at the **main** struct if we want to deprecate.

Reply:
ah right. thanks!